src/main/webapp/app/scores/exercise-scores-repo-export-dialog.component.ts

```diff
@@ -28,27 +42,38 @@ export class ExerciseScoresRepoExportComponent {
 
     exportRepos(exerciseId: number) {
         this.exportInProgress = true;
-        const studentIdList = this.studentIdList.split(',').map(e => e.trim());
-        this.exerciseService.exportRepos(exerciseId, studentIdList).subscribe(
-            response => {
-                this.jhiAlertService.success('Export of repos was successful. The exported zip file with all repositories is currently being downloaded');
-                this.activeModal.dismiss(true);
-                this.exportInProgress = false;
-                if (response.body) {
-                    const zipFile = new Blob([response.body], { type: 'application/zip' });
-                    const url = window.URL.createObjectURL(zipFile);
-                    const link = document.createElement('a');
-                    link.setAttribute('href', url);
-                    link.setAttribute('download', response.headers.get('filename')!);
-                    document.body.appendChild(link); // Required for FF
-                    link.click();
-                    window.URL.revokeObjectURL(url);
-                }
-            },
-            err => {
-                this.exportInProgress = false;
-            },
-        );
+        const studentIdList = this.studentIdList !== undefined && this.studentIdList !== '' ? this.studentIdList.split(',').map(e => e.trim()) : ['ALL'];
+        this.exerciseService
+            .exportRepos(
+                exerciseId,
+                studentIdList,
+                this.allStudents !== undefined ? this.allStudents : false,
```

Comment:
Again, if you would use an object for the settings, you would have less parameters in the class and you could easily set defaults.